### PR TITLE
Tests for addComment[UI/Server]

### PR DESCRIPTION
**Pull request by borgmaan**
- Adds tests to show comments can be added in the UI and show up appropriately in UI and DB after being submitted by user
- Moves data being used by `shinytest2` tests under `inst/testdata/` to align with pattern from other tests
- Also implicitly covers `R/viewComments.R` at 100% due to test design. `covr::package_coverage()` output shown below for reference. 

```
riskassessment Coverage: 59.04%
R/reportPreview.R: 18.83%
R/databaseView.R: 19.05%
R/reweightView.R: 20.66%
R/utils.R: 51.56%
R/communityMetrics.R: 56.90%
R/sidebar.R: 57.06%
R/utils_startup.R: 61.49%
R/run_app.R: 69.44%
R/app_server.R: 79.03%
R/assessmentInfo.R: 84.00%
R/utils_insert_db.R: 84.28%
R/uploadPackage.R: 86.25%
R/metricBox.R: 93.18%
R/utils_get_db.R: 95.12%
R/maintenanceMetrics.R: 97.56%
R/addComment.R: 100.00%
R/app_config.R: 100.00%
R/app_ui.R: 100.00%
R/introJS.R: 100.00%
R/metricGrid.R: 100.00%
R/viewComments.R: 100.00%
```

### CODE REVIEW

**Review by Jeff-Thompson12 — CHANGES_REQUESTED** [review]
@borgmaan Looks pretty good to me. In addition to the conversations below, could we add a few more items to test the addComment module more generally? Such as checking that `renderUI` is outputting the expected reactive text. We could also test that clicking submit when no comment has been entered doesn't do anything.

File: tests/testthat/test-addComment.R [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  # copy in already instantiated database to avoid need to rebuild
+  # this is a database that has been built via inst/testdata/upload_format.csv
+  test_db_loc <- system.file("testdata", "upload_format.database", package = "riskassessment")
+  file.copy(
+    test_db_loc,
+    app_db_loc
+  )
+
+  # confirm no comments exist in the database
+  con <- DBI::dbConnect(RSQLite::SQLite(), app_db_loc)
```
[the remark this answers is not in the document]
Comment (borgmaan) [reply tests/testthat/test-addComment.R:18]: Added logic to close connection in [9acbff4](https://github.com/pharmaR/riskassessment/pull/333/commits/9acbff4ad089c69708df71f56c885a6dc5050313).

File: tests/testthat/test-addComment.R [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  )
+  
+  # set up new app driver object
+  app <- AppDriver$new(app_dir = test_path("test-apps"))
+
+  # select dplyr package
+  app$set_inputs(`sidebar-select_pkg` = "dplyr")
+
+  # navigate to maintenance metrics tab
+  app$click(selector = "#tabs > li:nth-child(2) > a")
```
[the remark this answers is not in the document]
Comment (borgmaan) [reply tests/testthat/test-addComment.R]: Your suggestion is a much better way of doing this. Implemented in [f51a125](https://github.com/pharmaR/riskassessment/pull/333/commits/f51a1252741bddf0bea29020d3d22b8de19c84c0).

**Comment by borgmaan** [comment]
@Jeff-Thompson12 -- thanks for the feedback! I can certainly add a few more checks. 

>  Such as checking that renderUI is outputting the expected reactive text. 

Are you referring to user name/user role here? Happy to add checks for those.

**Comment by Jeff-Thompson12** [comment]
> Are you referring to user name/user role here? Happy to add checks for those.

Yes, the rendered UI is just a `textAreaInput`, but the label and placeholder depend upon arguments passed to the module.

**Comment by borgmaan** [comment]
@Jeff-Thompson12 -- I updated to add:

- A test to check that no action happens when Submit is pressed and the comment box is empty 
- Tests to confirm user and role are set appropriately in the comment 

Please let me know if there is anything else you'd like to see!

**Comment by Jeff-Thompson12** [comment]
@borgmaan I think we were talking past each other. I was specifically talking about the HTML elements `maintenanceMetrics-add_comment-add_comment` and `maintenanceMetrics-add_comment-add_comment-label`. Both of which are set in the `addComments` module [here](https://github.com/pharmaR/riskassessment/blob/fbde2d842fea5bb118990475df22a49da672102d/R/addComment.R#L42-L50).

**Comment by borgmaan** [comment]
@Jeff-Thompson12 you are right. My mistake. [038f8b1](https://github.com/pharmaR/riskassessment/pull/333/commits/038f8b1a1f7727aa6e58b144fd83b83bcd5d14a5) adds logic to test what you are referencing. 

The tests all are passing in the GHA workflows but the pkgdown build GHA keeps failing for me on [this line](https://github.com/pharmaR/riskassessment/blob/master/.github/workflows/pkgdown.yaml#L37). I also get an error if I try to run that install command locally. 

Perhaps something with the repo (it will run if I drop that option). I will retrigger again in the AM to see if it is resolved.

**Comment by borgmaan** [comment]
FYI -- pkgdown GHA issue is now resolved.

**Review by Jeff-Thompson12 — APPROVED** [review]
LGTM!

**Pull request merged**